mempool/clist_mempool.go

```diff
@@ -389,6 +389,18 @@ func (mem *CListMempool) resCbFirstTime(
 				return
 			}
 
+			// Check transaction not already in the mempool
+			if _, ok := mem.txsMap.Load(types.Tx(tx).Key()); ok {
```

Comment:
What would happen if the new request was checked at a different height, or it wants different amount of gas, or the sender is not the same as recorded in memTx? I think these values need to be updated in the mempool.

Reply:
Notice that this code is called as the callback for `CheckTx` and it is intended to add a new transaction to the mempool. It is not called by `CheckTx` in recheck mode, for instance, when the txs should be already in the mempool.

Reply:
PLus, updating the sender is handler elsewhere, namely here: https://github.com/cometbft/cometbft/blob/5f4a27eadb140825b260e8ddf22b5c3fb1820007/mempool/clist_mempool.go#L243

Reply:
You're right that the sender is recorded before sending the CheckTx request, so there's no need to update it here.

Reply:
But what about the height or the gas? 
Could it be that the second request was requested at a different height? In this case it would be better to update the height in `mempoolTx`. 
Is the gas the same even when rechecked at the same height? That is, should the gas calculation be deterministic?

Reply:
The senders are added where @cason pointed if the tx is in cache. But what if it is not in cache, because the cache overflows and thus this update  never happens? Then we need to retrieve the tx from the mempool at this point and update the senders here?

About the other concerns of @hvanz , I am not 100% sure how to handle it, but we definitely should either update it when retrieving the tx to update the senders, or ignore those values. But whichever we do, we need to document this somewhere so users are aware.

Reply:
Yes, I agree with you. But the  senders in particular are something we should update because this will prevent the node to gossip the transaction back to the sender. The mempool remembers all senders of a transaction (duplicate or not) . If the cache did not overflow, the sender would be added in the code Daniel posted. If it is not in the cache it would be added below the fix. RIght now , we should somehow add the sender within the fix as well (my opinion).

Reply:
Regarding the problem of not tracking the sender of a transaction properly, this is already the case in the initial codebase: if a transaction is received from some peer `p` before `resCbFirstTime` executes, the information about `p` is lost.

Reply:
After some discussion with Jasmina, I now agree that we should add senders to the existing entry here, as we do when the cache prevents us from checking a transaction again.

Reply:
I agree with updating the senders. In the current codebase, the sender is actually tracked when it is received either from a peer or via RPC, and before `resCbFirstTime`: https://github.com/cometbft/cometbft/blob/main/mempool/clist_mempool.go#L261

Reply:
Alright, I did so.